crates/client/src/lib.rs

```diff
@@ -310,109 +328,173 @@ impl<R: RegistryStorage, C: ContentStorage, N: NamespaceMapStorage> Client<R, C,
     pub async fn publish_with_info(
         &self,
         signing_key: &signing::PrivateKey,
-        mut info: PublishInfo,
+        publish_info: PublishInfo,
     ) -> ClientResult<RecordId> {
-        if info.entries.is_empty() {
+        if publish_info.entries.is_empty() {
             return Err(ClientError::NothingToPublish {
-                name: info.name.clone(),
+                name: publish_info.name.clone(),
             });
         }
 
-        let initializing = info.initializing();
-
         tracing::info!(
             "publishing {new}package `{name}`",
-            name = info.name,
-            new = if initializing { "new " } else { "" }
+            name = publish_info.name,
+            new = if publish_info.initializing() {
+                "new "
+            } else {
+                ""
+            }
         );
-        tracing::debug!("entries: {:?}", info.entries);
-
-        let package = match self.package(&info.name).await {
-            Ok(package) => {
-                if initializing {
-                    return Err(ClientError::CannotInitializePackage { name: package.name });
-                } else if info.head.is_none() {
-                    // If we're not initializing the package and a head was not explicitly specified,
-                    // updated to the latest checkpoint to get the latest known head.
-                    info.head = package.state.head().as_ref().map(|h| h.digest.clone());
+        tracing::debug!("entries: {:?}", publish_info.entries);
+
+        #[cfg(feature = "cli-interactive")]
+        let mut accepted_prompt_to_initialize = false;
+
+        let mut init_record_id: Option<RecordId> = None;
+
+        let (package, record) = loop {
+            let mut info = publish_info.clone();
+
+            #[cfg(feature = "cli-interactive")]
+            let mut initializing = info.initializing();
+
+            #[cfg(not(feature = "cli-interactive"))]
+            let initializing = info.initializing();
```

Comment:
This sort of thing is definitely not worth feature-gating. If the compiler complains:

```suggestion
            #[allow(unused_mut)] // feature = "cli-interactive"
            let mut initializing = info.initializing();
```

Reply:
fixed